packages/edit-site/src/components/sidebar-dataviews/default-views.js

```diff
@@ -26,18 +26,24 @@ import {
 	OPERATOR_IS_ANY,
 } from '../../utils/constants';
 
-export const DEFAULT_CONFIG_PER_VIEW_TYPE = {
-	[ LAYOUT_TABLE ]: {
-		primaryField: 'title',
-	},
-	[ LAYOUT_GRID ]: {
-		mediaField: 'featured-image',
-		primaryField: 'title',
-	},
-	[ LAYOUT_LIST ]: {
-		primaryField: 'title',
-		mediaField: 'featured-image',
-	},
+export const defaultFields = {
+	[ LAYOUT_TABLE ]: [
+		{ render: 'primary', field: 'title' },
```

Comment:
What I like about absorbing `primary`, `media`, etc. as render functions is that they become "UI slots". It's like saying: render this field in the _slot_ of the primary field.

They had other meanings (cannot be hidden, etc.), we need to consider whether those are well covered and still clear.

Reply:
Having looked a bit more into the details, I think using `render: primary` is conflating two concepts. Note that the fields still have to provide its own render function, even if they are the primary field. To me, we should only use `render: 'anything'` if that makes the field.render function redundant. For example, for combining fields, or when we have types so we can provide a default render function.

So, what about doing this instead:

```js
[
  { id: 'title', slot: 'primary' },
]
```

This has a couple of advantages:

- Conceptually, the purpose is clearer and it's clear how the `render` function works (the field still has to provide one).
- It also simplifies the code: we can lift up the `normalizeFieldRenderConfigs` to the DataViews component (shared across view types), and only work with `fields` internally — removing any reference to fieldsToRender. For example:

```js
// In dataviews.jsx

/*
 * - fields to render doesn't contain any field not to render (visible fields)
 * - at this point, it has also resolved any "preset render" (e.g.: render: 'my-custom-render', or type: 'date', etc.) into a function we can call.
 * - it preserves the slot: primary information
 */
const fieldsToRender = getFieldsToRender( fields, view.fields );
 
// Later on, in view-table.jsx

<TableRow
  fields={ fields }
  // Doesn't need the extra fieldsToRender
  />
```

Reply:
The `render` is not a slot for me and should't override the `field.render` for me at all. Here's how I see things:

 - The `render` is a render config for a field. Potentially it can container more configuration that are passed to the field render function later but it's not mandatory.
 - The field.render function is a function that is meant to render the base rendering of the field (without any decoration like making it primary, making it a badge...)
 - There can be `render` types that render multiple fields later (vertical, horizontal...) and these don't really represent slots.

I know that as things are implemented today, they can seem like "slots" (especially for media and badges). I agree that this abstraction is not perfect but I believe the slot abstraction won't cover the horizonal/vertical use-cases while the current one will.

I think there's room for improvement though and I would like to see how we should evolve this once we get a better feeling of it.

Reply:
Thinking more about this. I think if we take a look at the "list" and "grid" views, the media and primary fields there are clearly slots. The "badge" is more of a rendering config and "primary" in "tables" is kind of a rendering config too. When we'll introduce "horizontal" and "vertical" combination, these will be rendering configs for sure and not slots.

It's clear that there are two concepts that are overlapping each other and that the perfect solution is not clear enough. I'd like to go with a unique concept for now (rendering config) even if it's bended a little bit by the view components for now and see how far we can go with that.

Reply:
I agree with the direction, we just need more iterations to flesh out all the details.

I'd perhaps suggest renaming this prop to `renderAs: primary`, `renderConfig: primary`, etc. (a different name than `render` to prevent confusion with the `field.render` function).

Reply:
I also agree we have working with two concepts here: "slots" and "decorations/formats" to how a field is rendered. 

For awareness, I had [this PR](https://github.com/WordPress/gutenberg/pull/56942) a while where I explore the idea of "formats" to make the field.render function declarative. Perhaps we can revisit this idea with what we know today. Specially because the field render functions are coupled with the view types: see [this example](https://github.com/WordPress/gutenberg/blob/df2793df94c71a9c1f1f8312b86a73b9150760c5/packages/edit-site/src/components/posts-app/posts-list.js#L339) where we decorate the field with a link depending on the view type — that's another example that could be absorbed by "decorations/formats".

Reply:
I actually like the "format" name better than "render". The best name for me would be "type" actually, it's a type of rendering but it can be confusing with field types, so I'm probably switching to format for now.

Reply:
Format(s) sounds good to me as well 👍

Reply:
But then, if we go with formats, we should use it internally as well (e.g.: `fieldFormats = normalizeFieldFormats`, etc.) 🙈

Reply:
yes, although if we think about what "view.fields" today does (in this PR), it serves both as field formats but also as "visible fields". A lot of the tension is coming from this in all the places. Maybe we can just embrace "fieldFormats" in this PR and just continue iterating/polishing as we add the other formats, solve the reordering and the loss of formats...

Reply:
If you think about it, `view.fields` will represent the "template" of an item in the list view.

Reply:
Thinking more, I think `template` is the ideal world here and ideally its format is different between view types.

For tables it's just an array of fields, for list it's a `{ primary, media, otherFields }` for grid it's `{ primary, media, badgeFields, columnFields, rowFields }`. 

I think that's probably the ideal (different template shape per layout) but the fact that we toggle field visibility in the UI per field rather than as a template is what is making things hard.

That said, I'm going to stop bike-shedding right now and commit to the "format" we came up with for now so be able to follow-up with combination. I think we can revisit the template idea later.